File: programmers/모의고사 (2).py
def solution(answers):
    p1 = [1, 2, 3, 4, 5]
    p2 = [2, 1, 2, 3, 2, 4, 2, 5]
    p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    pattern = [p1, p2, p3]

    # 생각해보니까 굳이 각 학생의 배열은 만들어 줄 필요가 없다.
    # pattern이랑 answers 바로 비교 후 cnt 계산하기
    cnt = []
    for p in pattern:
        c = 0
        for i, a in enumerate(answers):
            if a == p[i % len(p)]:
                c += 1
        cnt.append(c)

    M = max(cnt)
    answer = []
    for i in range(len(cnt)):
        if cnt[i] == M:
            answer.append(i + 1)

    return answer

File: programmers/모의고사 (2).js
function solution(answers) {
  const p1 = [1, 2, 3, 4, 5];
  const p2 = [2, 1, 2, 3, 2, 4, 2, 5];
  const p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
  const pattern = [p1, p2, p3];

  let cnt = [];
  for (const p of pattern) {
    let c = 0;
    for (let i = 0; i < answers.length; i++) {
      if (answers[i] == p[i % p.length]) {
        c++;
      }
    }
    cnt.push(c);
  }

  M = Math.max(...cnt);
  var answer = [];
  for (let i = 0; i < cnt.length; i++) {
    if (cnt[i] == M) {
      answer.push(i + 1);
    }
  }

  return answer;
}
